src/platform/ASR/ASRFactoryDataParser.c

```diff
@@ -0,0 +1,308 @@
+/*
```

Comment:
why are we adding a C file in here? Matter SDK is generally C++.

Reply:
This file  is C++ now